lib/internal/crypto/random.js

```diff
@@ -281,10 +282,86 @@ function getRandomValues(data) {
   return data;
 }
 
+// Implements an RFC 4122 version 4 random UUID.
+// To improve performance, random data is generated in batches
+// large enough to cover kBatchSize UUID's at a time. The uuidData
+// and uuid buffers are reused for each. Each call to randomUUID()
+// consumes 16 bytes from the buffer.
+
+const kHexDigits = [
+  48, 49, 50, 51, 52, 53, 54, 55,
+  56, 57, 97, 98, 99, 100, 101, 102
+];
+
+const kBatchSize = 128;
+let uuidData;
+let uuid;
+let uuidBatch = 0;
+
+function randomUUID() {
+  if (uuidData === undefined) {
+    uuidData = new Uint8Array(16 * kBatchSize);
+    uuid = Buffer.alloc(36, '-');
+    uuid[14] = 52; // '4'
+  }
+
+  if (uuidBatch === 0) randomFillSync(uuidData);
+  const base = 16 * uuidBatch;
+  uuidBatch = (uuidBatch + 1) % kBatchSize;
+
+  // Variant byte: 10xxxxxx (variant 1)
+  uuidData[base + 8] = (uuidData[base + 8] & 0x3f) | 0x80;
```

Comment:
For the sake of consistency, I would personally find it better to read/understand if the version bit was also set in the `uuidData` instead of in the target `uuid` buffer.

After all there are two bits in the uuid which have to be set to specific non-random values, so I somewhat dislike the asymmetry in this code where one of the bits is set in the output buffer and one in the input buffer and then copied.

See https://github.com/uuidjs/uuid/blob/ed3240154759b748f6a3b7d545f3b10759ee4ba7/src/v4.js#L9-L11

Reply:
Oh I see now that the version bit is set only *once* for the entire process lifetime…

If this gives a perf benefit, so be it…